### PR TITLE
Added verbose mode closing #24

### PR DESCRIPTION
**Pull request by danieljuhl**

### CODE REVIEW

**Comment by danieljuhl** [comment]
@mafintosh I've justed made some changes - what do you think?

**Comment by mafintosh** [comment]
:+1:

File: index.js [hunk 1]
```diff
@@ -123,17 +124,18 @@ var compile = function(schema, cache, root, reporter, opts) {
...
     var indent = 0
     var error = function(msg) {
-      if (reporter === false) {
-        validate('errors++')
-        return
+      validate('errors++')
+      if (reporter === true) {
+        validate('if (validate.errors === null) validate.errors = []')
+        if (verbose) {
+          validate('validate.errors.push({field:"%s",message:"%s",value:%s})', formatName(name), msg, name)
```
Comment (ebdrup) [inline index.js]: Why is it that `field` is not allways the full path to the additional prop? (Showing the prop that is additional)

Our developers are really confused about this, even though we are using verbose mode.They thought they had an additinal property called `data`.

The `value` is not really a value, more a field name right?

**Comment by danieljuhl** [comment]
@ebdrup the root is always called `data`, as the name is unknown to the validator. If you read the error this way, I think it makes sense: "`field` has additionalProps: `value`", where field is your object, which does not validate, to follow the logic of the validator, that field is the field with error(s), and the value is the content of that field, in this specific case, it makes more sense to tell which fields is additional instead of outputting the whole object.

Or am I getting your question/comment wrong?

**Comment by ebdrup** [comment]
One could ague that the extra property of the object does not validate, in that is has a value, and it’s not supposed to.
But mainly, the other validators I’ve worked with output the additional property in `field` and that is much nicer to work with IMHO.
Especially when you have multiple additional properties that turn into many different errors. In the current implementation it’s just turned into similar error messages that `data` has additional properties. But the errors are actually different additional properties.

**Comment by danieljuhl** [comment]
Point taken...

**Pull request merged**